File: app/src/main/java/com/doubean/ford/util/SpanCountCalculator.java
package com.doubean.ford.util;

import android.content.Context;
import android.util.DisplayMetrics;

public class SpanCountCalculator {
    public static int getSpanCount(Context context, int minSpanWidthDps) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        float displayWidthDps = displayMetrics.widthPixels / displayMetrics.density;
        return (int) Math.ceil(displayWidthDps / minSpanWidthDps);
    }
}

File: app/src/main/java/com/doubean/ford/util/SpanCountCalculator.kt
package com.doubean.ford.util

import android.content.Context
import kotlin.math.ceil

object SpanCountCalculator {
    fun getSpanCount(context: Context, minSpanWidthDps: Int): Int {
        val displayMetrics = context.resources.displayMetrics
        val displayWidthDps = displayMetrics.widthPixels / displayMetrics.density
        return ceil((displayWidthDps / minSpanWidthDps).toDouble()).toInt()
    }
}